### ISSUE TITLE
Confusion about the vision_id_processor in the visual generation section

### ISSUE THREAD

**Issue by HalvesChen** [baaivision/UniVLA#37]
Thank you for your great work. 
I noticed that during visual generation, the `vision_id_processor `is called to process logits, setting logits outside the visual vocabulary to -inf. However, it seems that this part does not consider the `tokenizer.eof_token`. Based on how the training data of the dataset is constructed, a `tokenizer.eof_token` is added after each visual token in every line. From what I understand, this part also needs to be predicted by the model, right?

**Comment by Robertwyq** [comment]
@HalvesChen Hi,
During autoregressive generation, the model is restricted to generate only valid visual token IDs. Special tokens such as boi and eof are added externally via the prefix template (add_prefix_template).

https://github.com/baaivision/UniVLA/blob/a91ee5f269d7ad46d46635ee6481d2eb655b4af9/models/inference/inference_vision.py#L382

**Comment by HalvesChen** [comment]
@Robertwyq Thanks for your replay! 
I noticed that the token format for video and images seems to be different. The `to_imgstr()` adds the `tokenizer.eol_token` after each line, but the video does not. I would like to confirm if my understanding is correct, and whether the video processing format is Emu3's native format.  Also, can the` tokenizer.eol_token` be simply dropped from the imgstr after fine-tuning?

**Comment by Robertwyq** [comment]
@HalvesChen Your understanding is correct — the native version of Emu3 likely includes the EOL token, but its impact is minimal. These special tokens can be removed during the finetuning stage.
